src/path/ft.rs

```diff
@@ -59,6 +72,17 @@ impl FileTypePredicate {
         self.follow = yes;
         self
     }
+
+    pub(crate) fn new(path: &path::Path) -> io::Result<FileTypePredicate> {
```

Comment:
Would it be useful to make this `pub`?

What about naming it `from_path` or `with_path`?